src/components/__tests__/IspSelectWithIncidents-test.jsx

```diff
@@ -0,0 +1,139 @@
+          const ispId = ispIds[i];
+          const existsInSelected = selected.some(isp => isp.client_asn_number === ispId);
+          if (!existsInSelected) {
+            selected.push(isps[ispId]);
+          }
+        }
+      }
+
+      const incidentData = {};
+
+      const onChangeIncidentASN = function () { return; };
+
+      const wrapper = mount(
+        <IspSelectWithIncidents
+          incidentData={incidentData}
+          isps={isps}
+          onChangeIncidentASN={onChangeIncidentASN}
+          selected={selected}
+          onChange={mockChange}
+        />
+      );
+
+      const instance = wrapper.instance();
+
+      // Call select function on AT&T
+      instance.onAdd(att);
+
+      // Make sure that an item has been added to parent's 'selected' member
+      expect(selected).to.eql([att]);
+
+      // Add Verizon
+      instance.onAdd(verizon);
+
+      // Expect Verizon and AT&T to be in selected
+      expect(selected).to.eql([att, verizon]);
+
+      wrapper.unmount();
+    });
+
+    it('removing multiple ISPs', () => {
+      const selected = [att, verizon, twc];
+
+      function mockChange(ispIds) {
+        // ISP is deleted if in selected but not passed into this function
+        for (let i = 0; i < selected.length; i++) {
+          const isp = selected[i];
+          if (!ispIds.includes(isp.client_asn_number)) {
+            selected.splice(i, 1);
+          }
+        }
+      }
+
+      const incidentData = {};
+
+      const onChangeIncidentASN = function () { return; };
+
+      const wrapper = mount(
+        <IspSelectWithIncidents
+          incidentData={incidentData}
+          isps={isps}
+          onChangeIncidentASN={onChangeIncidentASN}
+          selected={selected}
+          onChange={mockChange}
+        />
+      );
+
+      const instance = wrapper.instance();
+
+      // Remove AT&T from selected
+      instance.onRemove(att);
+
+      // Make sure that AT&T was removed
+      expect(selected).to.eql([verizon, twc]);
+
+      // Remove Verizon from selected
+      instance.onRemove(verizon);
+
+      // Make sure that Verizon was removed
+      expect(selected).to.eql([twc]);
+
+      wrapper.unmount();
+    });
+
+    it('check if ISPs with incidents have an incident tooltip and an incident viewer button', () => {
+      const incidentData = {
+        AS11486x: [{}, {}], // Verizon with 2 incidents
+        AS10774x: [{}],  // AT&T with one incident
+      };
+      const selected = [];
+
+      const wrapper = shallow(
+        <IspSelectWithIncidents
+          incidentData={incidentData}
+          isps={isps}
+          selected={selected}
+        />
+      );
+
+      // There should be two incident tips because of the two different ISPs
+      expect(wrapper.find('#incident-isp-tip')).to.have.lengthOf(2);
+
+      // Both the row itself and the 'Show Incident' button have the same id,
+      // so each ISP with an incident should have two instances of their id.
```

Comment:
It is unclear to me why the 'Show Incident' button has an identical id to the ASN. Is this an intentional design choice?